Fix part of #10306: Added async keyword to asynchronous functions that return a promise

kendallpomerleau opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #10306.
2. This PR does the following: Adds the async keyword to the createTopic function, changes function name to createTopicAsync, and updates all references to that function.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

Commit:
added async keywords

kendallpomerleau commented:
@nithusha21 I'm not sure how/if I have to add a label, but I've been working on this as part of Grace Hopper Open Source Day and just wanted to notify someone. Thanks!

EricZLou reviewed (approved):
Looks good to me! Congrats on the PR, Kendall :)

kendallpomerleau commented:
@nithusha21 @Showtim3

srijanreddy98 reviewed (commented):
LGTM! -- thanks @kendallpomerleau.

Showtim3 reviewed (approved):
Looks awesome.
Thanks :D

seanlip commented:
The tests for this PR are consistently failing, so I'm not convinced it actually works. @kendallpomerleau, did you test this PR manually? (E.g. did you try to create a topic?)

kendallpomerleau commented:
I did test it, and as far as I can tell, it works. I only changed 3 lines and I went off of a previous PR that was successfully merged, so I'm not sure why tests are failing.

seanlip commented:
Could you explain how you tested it? (Sorry to ask for more detail, just wanted to make sure we're on the same page about that.)

kendallpomerleau commented:
On the dev Oppia, I created a new exploration, added an introduction, added some interactions, then saved and published. I couldn't find a way to create a topic other than that. If you could point me in a different direction if that is not correct, that'd be great.

seanlip commented:
Ah, yup! You're creating an exploration, which is not the same as a topic. 

Making a topic is a bit more involved. To do that:

1. Log into your dev server as an admin.
2. Go to /admin, go to Roles, set your role to Admin. (You can find your username in the dropdown on the top right.)
3. Go to /topics-and-skills-dashboard (it should show up in your top-right menu when you go back to the main page of the site).

You should be able to create a topic from there. Hope that helps!

DubeySandeep commented:
@kendallpomerleau, any update on this PR? Do let us know if you still facing any issue or need any help., Thanks! :)
